src/js/internal/webstreams/adapters.js

```diff
@@ -0,0 +1,1106 @@
+const {
+  Symbol,
+  TypedArrayPrototypeGetBuffer,
+  TypedArrayPrototypeGetByteOffset,
+  TypedArrayPrototypeGetByteLength,
+  SafePromisePrototypeFinally,
+  StringPrototypeToLowerCase,
+  SafePromiseAll,
+  PromisePrototypeThen,
+  PromiseResolve,
+} = require("internal/primordials");
+
+const { Stream } = require("../streams/legacy");
+const { Buffer } = require("node:buffer");
+
+const eos = require("../streams/end-of-stream");
+
+const { isDestroyed, isReadable, isWritable } = require("../streams/utils");
+
+const { AbortError } = require("../../node/events");
+const { validateObject, validateBoolean } = require("../validators");
+
+const { createDeferredPromise } = require("../../node/util");
+
+const { Readable } = require("../streams/readable");
+
+const { isReadableStream, isWritableStream } = require("../streams/utils");
+
+const Duplex = require("../streams/duplex");
+
+const kEmptyObject = Object.freeze({ __proto__: null });
+
+const { destroy } = require("../streams/destroy");
+
+function handleKnownInternalErrors(cause) {
+  switch (true) {
+    case cause?.code === 'ERR_STREAM_PREMATURE_CLOSE': {
+      return new AbortError(undefined, { cause });
+    }
+    // case ZLIB_FAILURES.has(cause?.code): {
+    //   // eslint-disable-next-line no-restricted-syntax
+    //   const error = new TypeError(undefined, { cause });
+    //   error.code = cause.code;
+    //   return error;
+    // }
+    default:
+      return cause;
+  }
+}
+
+function normalizeEncoding(enc) {
```

Comment:
Can we delete this? and `slowCases`? I believe we already have this code somewhere in both C++ and Zig